### PR TITLE
feat(query): support SHOW DROP DATABASES

### PR DESCRIPTION
**Pull request by TCeason**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

Support `SHOW DROP DATABASES [FROM <database>] [<show_limit>]`

Add new system table databases_with_history

Fix: #16801

## Tests

- [x] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/databendlabs/databend/16811)

### CODE REVIEW

File: src/query/storages/iceberg/src/catalog.rs [hunk 1]
```diff
@@ -222,6 +222,10 @@ impl Catalog for IcebergCatalog {
         Ok(Arc::new(IcebergDatabase::create(self.clone(), db_name)))
     }
 
+    async fn list_databases_history(&self, _tenant: &Tenant) -> Result<Vec<Arc<dyn Database>>> {
+        unimplemented!()
```
Comment (TCeason) [inline src/query/storages/iceberg/src/catalog.rs:226]: Could I make this func with unimplemented? cc @Xuanwo

**Review by andylokandy — APPROVED** [review]

File: src/query/storages/hive/hive/src/hive_catalog.rs [hunk 2]
```diff
@@ -293,6 +293,11 @@ impl Catalog for HiveCatalog {
         Ok(res)
     }
 
+    async fn list_databases_history(&self, _tenant: &Tenant) -> Result<Vec<Arc<dyn Database>>> {
+        // TODO: Implement list_databases_history
+        unimplemented!()
```
[the remark this answers is not in the document]
Comment (Xuanwo) [reply src/query/storages/hive/hive/src/hive_catalog.rs:298]: It's fine for now.

**Review by b41sh — APPROVED** [review]

**Review by b41sh — APPROVED** [review]

**Review by b41sh — APPROVED** [review]

**Comment by TCeason** [comment]
Please merge this pr. @BohuTANG

**Comment by TCeason** [comment]
Cc @soyeric128 doc needed

**Pull request merged**